File: Android/VKAudioPlayer/app/src/main/java/github/y0rrrsh/vkaudioplayer/utils/VKAPUtils.java
package github.y0rrrsh.vkaudioplayer.utils;

import android.app.Activity;
import android.content.Context;

import github.y0rrrsh.vkapi.VKApi;
import github.y0rrrsh.vkaudioplayer.activities.LoginActivity;
import github.y0rrrsh.vkaudioplayer.database.vkitem.VkItemDB;

import static github.y0rrrsh.vkaudioplayer.BuildConfig.VKAP_APP_ID;
import static github.y0rrrsh.vkaudioplayer.BuildConfig.VKAP_APP_SCOPE;

/**
 * @author Artur Yorsh. 07.06.16.
 */
public class VKAPUtils {

    public static String formatProgress(int progress) {
        int minutes = progress / 60;
        int seconds = progress % 60;

        return String.format("%02d:%02d", minutes, seconds);
    }

    public static boolean lastRequestIsOlder(Context context, String dataTag, double minutes) {
        long lastDataUpdate = VKAPPreferences.getLastDataUpdate(context, dataTag);
        long now = System.currentTimeMillis();

        return now - lastDataUpdate > minutes * 60 * 1000;
    }

    public static void resetSettings(Context context) {
        VKAPPreferences.clear(context);
        VkItemDB.getInstance().setSyncForAllEnabled(false);
    }

    public static void login(Activity activity) {
        LoginActivity.start(activity, VKAP_APP_ID, VKAP_APP_SCOPE);
    }

    public static void logout(Activity activity) {
        VKApi.logout(activity);
    }
}

File: iOS/VkAudio/utils/VKAPUtils.swift
//
//  VKAPUtils.swift
//  VkAudio
//
//  Created by mac-224 on 22.06.16.
//  Copyright © 2016 y0rrrsh. All rights reserved.
//

import Foundation

class VKAPUtils {

    static let REQUEST_DELAY_USER_AUDIOS = 15
    static let REQUSET_DELAY_USER_GROUPS = 5 * 60
    static let REQUEST_DELAY_USER_FRIENDS = 5 * 60
    
    static func formatProgress(seconds: Int) -> String {
        let minutes = seconds / 60
        let seconds = seconds % 60
        
        return String(format: "%0.2d:%0.2d", minutes, seconds)
    }
    
    static func lastRequestIsOlder(dataTag: String, seconds: Int) -> Bool {
        let lastDataUpdate = VKAPUserDefaults.getLastDataUpdate(dataTag)
        let now = NSDate.currentTimeMillis()
        
        return now - lastDataUpdate > Double(seconds) * 1000
    }
    
    static func resetSettings() {
        VKAPUserDefaults.clear();
        VkModelDB.sharedInstance!.setSyncForAllEnabled(false);
    }
    
    static func login(starter: UIViewController) {
        let plistPath = NSBundle.mainBundle().pathForResource("VKAPApplication", ofType: "plist")
        let appValues = NSDictionary(contentsOfFile: plistPath!)!
        
        let appId = appValues.objectForKey("APP_ID") as! String
        let appScope = appValues.objectForKey("APP_SCOPE") as! String
        
        VKApi.login(starter, appId: appId, appScope: appScope)
    }
    
    static func logout(holder: UIViewController) {
        VKApi.logout(holder)
        let avatarName = "bg_image_\(VKApi.userId!).jpg"
        LocalStorage.removeFile(.DocumentDirectory, fileName: avatarName)
        
        login(holder)
    }
}